File: Python/0011_container_with_most_water/0011_container_with_most_water.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        # Two Pointers Traverse
        left, right, maximum = 0, len(height) - 1, 0
        while left < right:
            maximum = max(maximum, min(height[left], height[right]) * (right - left))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return maximum

File: Java/0011_container_with_most_water/0011_container_with_most_water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, maximum = 0;
        while (left < right) {
            maximum = Math.max(maximum, Math.min(height[left], height[right]) * (right - left));
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return maximum;
    }
}